docs/language-server/protocol-project-manager.md

```diff
@@ -335,6 +336,27 @@ null;
 
 #### Errors
 
+- [`ProjectDataStoreError`](#projectdatastoreerror) to signal problems with
+  underlying data store.
+
+### Write to File
+
+Writes bytes from stdin to the provided path.
+
+#### Parameters
+
+```typescript
+echo 'Hello World!' | project-manager --filesystem-write-path {path}
```

Comment:
Is this safe against path traversal attacks? Can path be relative or absolute or both? I don't think we check if the user can call `../../..` etc in the path.

Reply:
The OS is responsible for checking the permissions when the process tries to access the filesystem. In this case, we allow writing to any file that has appropriate permissions.
We don't expose this endpoint to the running project-manager process through JSON-RPC protocol. This and other filesystem-related methods can only be called through the command line.